airbyte-commons-worker/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -534,11 +535,9 @@ public KubePodProcess(final boolean isOrchestrator,
         .withLabels(labels)
         .withAnnotations(annotations)
         .endMetadata()
-        .withNewSpec();
-
-    if (isOrchestrator) {
-      podBuilder = podBuilder.withServiceAccount("airbyte-admin").withAutomountServiceAccountToken(true);
-    }
+        .withNewSpec()
+        .withServiceAccount(isOrchestrator ? "airbyte-admin" : serviceAccount)
```

Comment:
@benmoriceau [added a comment to the previous PR](https://github.com/airbytehq/airbyte/pull/19010/files#r1046268059):

> Should we add a dedicated variable for the orchestrator SA and well as the one added in order to have something like:
> `.withServiceAccount(isOrchestrator ? orchestratorServiceAccount : serviceAccount)`. If we go that way, we need to make sure that the default value is `"airbyte-admin"`.